src/derivable/data-source.ts

```diff
@@ -0,0 +1,223 @@
+import { isRecordingObservations, recordObservation } from '../tracking';
```

Comment:
De `data-source` is nu heel erg 'pull-based' opgezet. Als iemand vraagt om een nieuwe waarde, dan kan deze berekend worden. En eventueel kan er iets gezet worden van buitenaf.

Maar in mijn ogen is het idee van een `data-source` meer dat deze de bron van data kan zijn (vanuit events/messages etc). En dat zonder dat de `data-source` zelf settable moet zijn.

Met de huidige implementatie kan dit, maar is het zeker niet triviaal.

Reply:
I don't agree that DataSource is pull-based. It is lazy, so `calculateCurrentValue` is only called when necessary, but the implementer indicates using `checkForChanges` that a new value is available. The whole idea is to facilitate situations where you want to connect and disconnect based on whether the DataSource is observed or not. Otherwise, you could simply use an Atom.